### ISSUE TITLE
Ho to implement two form control into  single formely fields ?

### ISSUE THREAD

**Issue by mihirhapaliya** [formly-js/ngx-formly#1222]
I want to place  two controls into single, one is textbox and another dropdown is dropdown.

for ex. textbox is name and dropdown is profession.
The custom control should set and get a value in the form of object {name: 'abc' , profession: 'xyz'} .
I want to set a single label for both of the controls.

![custom-control](https://user-images.githubusercontent.com/32834137/46509396-0587c680-c860-11e8-86f6-255abf549074.png)

**Comment by aitboudad** [comment]
well there are several ways, here is one of them https://stackblitz.com/edit/angular-xm4ucn

**Comment by mihirhapaliya** [comment]
thanks . this really helps me. 

How can I apply validations on the controls.
for e.g. Name and profession should be required fields.
In other scenario I have number field instead of name , then how can I apply validation on the same ?

**Comment by aitboudad** [comment]
> How can I apply validations on the controls.

add `required` to the `defaultOptions` defined `custom` field definition:
```ts
      types: [
        { 
          name: 'custom', 
          component: FormlyFieldCustomInput, 
          wrappers: ['form-field'],
          defaultOptions: {
            fieldGroup: [
              { type: 'input', key: 'name', templateOptions: { required: true } },
              { type: 'input', key: 'profession', templateOptions: { required: true } },
            ]
          }
        },
      ],
```

for more advanced usage of validation check the demo examples

**Comment by mihirhapaliya** [comment]
thanks a lot.

Last question.
How can assign the values  to controls initially ?
for ex.
if assign model = { "custom": { "name": "mihir", "profession": "2" } }
Then it should set the value Mihir to name input  and select "Two" in select control

**Comment by aitboudad** [comment]
see https://formly-js.github.io/ngx-formly/examples/field-options/default-value

**Comment by mihirhapaliya** [comment]
Hi ,

Validation message is not displayed through I have given validation messages . Please help
 
types: [
    {
        name: 'custom',
        component: FormlyFieldCustomInput,
        wrappers: ['form-field'],
        defaultOptions: {
            fieldGroup: [
                {
                    type: 'input', key: 'name', templateOptions: { required: true }, validation: {
                        messages: {
                            required: 'required',
                        },
                    },
                },
                {
                    type: 'input', key: 'profession', templateOptions: { required: true }, validation: {
                        messages: {
                            required: 'required',
                        },
                    },
                },
            ]
        }
    },
],

**Comment by aitboudad** [comment]
you have to replicate your example using stackblitz in order to check the issue.

**Comment by mihirhapaliya** [comment]
here is the example. 
https://stackblitz.com/edit/angular-howafa
validations message is not shown , but its performed well

**Comment by aitboudad** [comment]
here is another better way https://stackblitz.com/edit/angular-cshif3 without using `custom-input.component`

**Comment by mihirhapaliya** [comment]
but I need to pass select options from service ,  So in 1st example I can pass through templateoptions and can bind that value to template . 
Is there way to solve the issue using custom component ?
Please help me with that if possible .
Can we  set validator to form control in ngonInit() ? I am not familier much in creating custom type . Please help.

**Comment by aitboudad** [comment]
then use a custom validator instead of relying to the required field

**Comment by eng-dibo** [comment]
> well there are several ways, here is one of them https://stackblitz.com/edit/angular-xm4ucn

this causes an error in Angular v14
```
'AbstractControl<any, any> | null' is not assignable to type 'FormControl<any>'.
```

**Comment by aitboudad** [comment]
> this causes an error in Angular v14
> 'AbstractControl<any, any> | null' is not assignable to type 'FormControl<any>'.

see https://github.com/ngx-formly/ngx-formly/issues/2842#issuecomment-1016476706

**Comment by KakashiOfAnbu** [comment]
In Formly 6.1.1, the first implementation doesn't seem to work since it doesn't update the model after the fields change. Here is the stackblitz: https://stackblitz.com/edit/angular-ivy-rabcxg?file=src/app/app.component.html. I am thankful if you can provide an alternative or some ways to address that behavior.

**Comment by aitboudad** [comment]
@KakashiOfAnbu its a bug, you may check https://github.com/ngx-formly/ngx-formly/issues/3455#issuecomment-1281110976 for a workaround

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.1.7](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.7) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
